### PR TITLE
home_work6

### PR DESCRIPTION
**Pull request by Lytvynenko18**
It's my first time with js functions, so it took a lot of time to find out how it works.
Pls reveiw.

### CODE REVIEW

File: homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    if (!passwordLength(password)) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Password should from 8 to 12 chars!' });
+    }
+
+    if (!checkbox) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Check the box that you are not a robot!' });
+    }
+
+    if (isEmail(validEmail) && passwordLength(password) && checkbox) {
```
Comment (vladyslav-yermolin-moc) [inline homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js:81]: You make the same operations here for the second time.
To not violate the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) you can refactor you code:

```
declare an empty errors object
if (email is not valid) {
errors[email_id] = 'text of error'
}
... // same for other errors

if (error object has any key) {
  setErrors(errors object)
}

```

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]

**Comment by vladyslav-yermolin-moc** [comment]
Hey, @Lytvynenko18, do you plan to fix my comments here?
I intend to close this pr otherwise.

File: homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Style/login.css [hunk 2]
```diff
@@ -146,3 +146,8 @@ button:hover {
   margin-left: 30%;
   margin-right: auto;
 }
+
+.errors {
```
[the remark this answers is not in the document]
Comment (Lytvynenko18) [reply homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Style/login.css:150]: Hey, @vladyslav-yermolin-moc sorry for so huge delay with answer. I've had the craziest 2 weeks of my entire year.
This is fixed in bbc401a62ebc25e07e1983dd967a0f7eee76836d

File: homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+function validateForm() {
+    deleteErrors();
+    const email = getValueById(EMAIL_INPUT_ID);
+    const password = getValueById(PASSWORD_INPUT_ID);
+    const checkbox = getValueById(NOT_A_ROBOT_CHECKBOX_ID);
+    const validEmail = deleteSpaces(email);
+
+    if (!isEmail(validEmail)) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Email invalid!' });
```
[the remark this answers is not in the document]
Comment (Lytvynenko18) [reply homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js]: Fixed in a78935e956123d2a83614ad7c3b87f34e7583509

File: homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js (hunk 1 above)
Comment (Lytvynenko18) [reply homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js:81]: I don't get it what U mean?
I wrote sequential checks for each field. If all data are correct, redirect to the "Success" page.
Could U please give me some details here,?

Comment (vladyslav-yermolin-moc) [reply homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js:81]: I mean, you call `isEmail` and `passwordLength(password)` twice. And you call `setErrors` 3 times for each error. 
With this aproach, which I've described [here](https://github.com/MastersAcademy/fe-course-basic-2023/pull/314#discussion_r1393869160) you would not repeat yourself and call `isEmail` and `passwordLength(password)` only 1 time. Same for `setErrors` - you would call it once with whole bunch of errors

Comment (vladyslav-yermolin-moc) [reply homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js:81]: But that is ok for now, your code is working properly, and my comments are related to refactoring and optimisation only. So I will merge it.

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Comment by Lytvynenko18** [comment]
Hey, Vlad.
Thank's a lot. I'll try to refactor code in addition to your comment.

**Pull request merged**